src/main/java/org/walkmod/javalang/compiler/reflection/ResultBuilderFromGenerics.java

```diff
@@ -71,10 +71,8 @@ public SymbolTable build(SymbolTable genericsSymbolTable) throws Exception {
 
 		if (generics != null) {
 			SymbolType[] syms = ASTSymbolTypeResolver.getInstance().valueOf(generics);
-			SymbolType scope = new SymbolType();
-			scope.setParameterizedTypes(Arrays.asList(syms));
 			genericsSymbolTable.pushScope();
-			updateTypeMapping(method.getGenericReturnType(), genericsSymbolTable, scope, true, new HashSet<String>());
+			updateTypeMappings(Arrays.asList(syms), method.getTypeParameters(), true, genericsSymbolTable, new HashSet<String>());
```

Comment:
The fixing comes from `method.getGenericReturnType` instead of `method.getTypeParameters` ?